### ISSUE TITLE
Pass false in UserMenu prop in AppBar component

### ISSUE THREAD

**Issue by mariaefi29** [marmelab/react-admin#5419]
**Is your feature request related to a problem? Please describe.**

Sometimes applications are not required to have any users. 

For that purpose I think it would be a great idea to pass `userMenu={false}` as a prop in a custom AppBar component.

**Describe the solution you'd like**

It would be great to be able to create your own custom AppBar component very easy like this:

`const MyAppBar = props => <AppBar {...props} userMenu={false} />;`

**Describe alternatives you've considered**

Now there is an error if your are trying to pass false, because userMenu requires JSX.Element:

`userMenu?: JSX.Element; `

There is a workaround to pass icon={false} in a custom UserMenu component and you will not see the icon of the profile page. 

```
const MyUserMenu = props => <UserMenu {...props} icon={false} />;

const MyAppBar = props => <AppBar {...props} userMenu={<MyUserMenu />} />;
```

However, the component is still there and clickable:

![image](https://user-images.githubusercontent.com/25860669/96412751-4195e600-11f3-11eb-8c93-34b742485de4.png)

Are there any other alternatives that I probably haven't found?

**Comment by fzaninotto** [comment]
Fixed by #5421

**Comment by mariaefi29** [comment]
Dear @fzaninotto,

Could you please provide some informations when this fix will be released?)

Thank you!

**Comment by djhi** [comment]
Hi @mariaefi29, as it has been merged on the `next` branch, it will be released in `3.10.0`. As for the date of the release, we can't really say. **Should** be in a few weeks.

**Comment by mariaefi29** [comment]
@djhi Thank you very much for your reply! I'll keep waiting then for 3.10.0 :)
